lib/container/ContainerEntryModule.js

```diff
@@ -0,0 +1,193 @@
+	 */
+	readableIdentifier(requestShortener) {
+		return `container entry`;
+	}
+
+	/**
+	 * @param {NeedBuildContext} context context info
+	 * @param {function(WebpackError=, boolean=): void} callback callback function, returns true, if the module needs a rebuild
+	 * @returns {void}
+	 */
+	needBuild(context, callback) {
+		return callback(null, !this.buildMeta);
+	}
+
+	/**
+	 * @param {WebpackOptions} options webpack options
+	 * @param {Compilation} compilation the compilation
+	 * @param {ResolverWithOptions} resolver the resolver
+	 * @param {InputFileSystem} fs the file system
+	 * @param {function(WebpackError=): void} callback callback function
+	 * @returns {void}
+	 */
+	build(options, compilation, resolver, fs, callback) {
+		this.buildMeta = {};
+		this.buildInfo = {
+			strict: true
+		};
+
+		this.clearDependenciesAndBlocks();
+
+		for (const dep of this.expose) {
+			const block = new AsyncDependenciesBlock(
+				undefined,
+				dep.loc,
+				dep.userRequest
+			);
+			block.addDependency(dep);
+			this.addBlock(block);
+		}
+
+		callback();
+	}
+
+	/**
+	 * @param {CodeGenerationContext} context context for code generation
+	 * @returns {CodeGenerationResult} result
+	 */
+	codeGeneration({ moduleGraph, chunkGraph, runtimeTemplate }) {
+		const sources = new Map();
+		const runtimeRequirements = RUNTIME_REQUIREMENTS;
+		const getters = [];
+
+		for (const block of this.blocks) {
+			const {
+				dependencies: [dep]
+			} = block;
+			const name = dep.exposedName;
+			const mod = moduleGraph.getModule(dep);
+			const request = dep.userRequest;
+
+			let str;
+
+			if (!mod) {
+				str = runtimeTemplate.throwMissingModuleErrorBlock({
+					request: dep.userRequest
+				});
+			} else {
+				str = `return ${runtimeTemplate.blockPromise({
+					block,
+					message: request,
+					chunkGraph,
+					runtimeRequirements
+				})}.then(${runtimeTemplate.basicFunction(
+					"",
+					`return ${runtimeTemplate.moduleRaw({
+						module: mod,
+						chunkGraph,
+						request,
+						weak: false,
+						runtimeRequirements
+					})}`
+				)});`;
+			}
+
+			getters.push(
+				`${Template.toNormalComment(
+					`[${name}] => ${request}`
+				)}"${name}": ${runtimeTemplate.basicFunction("", str)}`
+			);
+		}
+
+		sources.set(
+			"javascript",
+			new ConcatSource(
+				`\n${
+					runtimeTemplate.supportsConst() ? "const" : "var"
+				} __MODULE_MAP__ = {${getters.join(",")}};`,
+				`\n${
+					runtimeTemplate.supportsConst() ? "const" : "var"
+				} __GET_MODULE__ = ${runtimeTemplate.basicFunction(
+					["module"],
+					`return typeof __MODULE_MAP__[module] ==='function' ? __MODULE_MAP__[module].apply(null) : Promise.reject(new Error('Module ' + module + ' does not exist.'))`
+				)};`,
+				`\n\n${RuntimeGlobals.definePropertyGetters}(exports, {\n`,
+				Template.indent([
+					`get: ${runtimeTemplate.basicFunction("", "return __GET_MODULE__")},`,
+
+					`override: ${runtimeTemplate.basicFunction(
```

Comment:
Maybe remove this from the commit, as it's not finished yet.

It can be added in a later commit, so that this first commit is complete in itself.

Reply:
Yeah good point